model/src/main/kotlin/utils/Extensions.kt

```diff
@@ -208,3 +212,49 @@ fun List<ScanResult>.filterByProject(project: Project): List<ScanResult> {
         }
     }
 }
+
+/**
+ * Filter [packages][this] to include only [Package]s from [categoriesToInclude]. License choices from [ortResult] are
+ * respected and applied to dismiss licenseCategories that that are not chosen.
+ */
+fun List<Package>.filterByLicenseCategories(
+    categoriesToInclude: Collection<String>,
+    licenseCategorizations: List<LicenseCategorization>,
+    ortResult: OrtResult
+): List<Package> {
```

Comment:
I'm not super happy about this method, it isn't quite generic enough to justify a public extension function.
However I wanted to have it tested and adding it to `DownloaderCommand` as an `internal` method just to be able to test it seemed also wrong.

Passing the `ortResult` to implicitly apply the license choices is also not very nice. This could be changed by moving the responsibilty to the caller:
```kt
fun List<Package>.filterByLicenseCategories(
    categoriesToInclude: Collection<String>,
    licenseCategorizations: List<LicenseCategorization>,
    licenseInfoResolver: LicenseInfoResolver,
    repositoryLicenseChoices: List<LicenseChoice> = emptyList(),
    packageLicenseChoices: List<PackageLicenseChoice> = emptyList()
)
```

What do you think?

Reply:
As an idea, maybe this could be an extension function of `OrtResult` instead, in case it is applied to all packages from the result anyway: `OrtResult.getPackagesWithLicenseCategories(...)`
I think such a function would make sense in the public API even if it's (so far) used only once by ORT itself.

Reply:
Due to this suggestion: https://github.com/oss-review-toolkit/ort/pull/3935#discussion_r626502111
This extension is no longer required.